daemon/api_notices.go

```diff
@@ -173,12 +183,41 @@ func sanitizeTypesFilter(queryTypes []string) ([]state.NoticeType, error) {
 		}
 		types = append(types, noticeType)
 	}
-	if len(types) == 0 && len(typeStrs) > 0 {
-		return nil, errors.New("all requested notice types invalid")
+	if len(types) == 0 {
+		if len(typeStrs) > 0 {
+			return nil, errors.New("all requested notice types invalid")
+		}
+		// No types were specified, populate with notice types snap can view
+		// with its connected interface.
+		ucred, iface, err := ucrednetGetWithInterface(r.RemoteAddr)
+		if err != nil {
+			return nil, err
+		}
+		if ucred.Socket != dirs.SnapSocket {
```

Comment:
Would it be better to change this to and explicit check of `ucred.Socket == dirs.SnapdSocket` instead since what if a 3rd snapd socket gets added in the future - it would then be granted access here.

Reply:
Updated